src/main/scala/chisel3/stage/ChiselAnnotations.scala

```diff
@@ -99,6 +99,25 @@ case object WarningsAsErrorsAnnotation
 
 }
 
+/** Root directory for source files, used for enhanced error reporting */
+case class SourceRootAnnotation(directory: File) extends NoTargetAnnotation with Unserializable with ChiselOption
+
+object SourceRootAnnotation extends HasShellOptions {
+  val options = Seq(
+    new ShellOption[String](
+      longOption = "source-root",
+      toAnnotationSeq = { dir =>
+        val f = new File(dir)
+        if (!f.isDirectory()) {
+          throw new OptionsException(s"Must be directory that exists!")
+        }
+        Seq(SourceRootAnnotation(f))
+      },
+      helpText = "Root directory for source files, used for enhanced error reporting"
+    )
+  )
```

Comment:
Is it reasonable that this could be a `Seq[String]` instead of a single value? I.e., it's not just a single project root, but could be a bunch of directories where sources live. (The option on the `firtool` side accepts a list.)

Reply:
I should be able to make that work!